spec/integration/provider/image_streamer_deployment_group_spec.rb

```diff
@@ -0,0 +1,77 @@
+################################################################################
+# (C) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+require 'spec_helper'
+
+provider_class = Puppet::Type.type(:image_streamer_deployment_group).provider(:image_streamer)
+
+describe provider_class do
+  let(:resource) do
+    Puppet::Type.type(:image_streamer_deployment_group).new(
+      name: 'deployment-group-1',
+      ensure: 'found',
+      data: {
+        'name' => 'Absent-Group'
+      }
+    )
+  end
+
+  let(:provider) { resource.provider }
+
+  let(:instance) { provider.class.instances.first }
```

Comment:
instance declared but no tests are using it. Should have a test to make sure self.instances is able to run correctly for the resource. Something like
```ruby
it 'should be able to run through self.instances' do
  expect(instance).to be
end
```